agent/engine/docker_task_engine.go

```diff
@@ -1238,12 +1240,139 @@ func (engine *DockerTaskEngine) GetDaemonManagers() map[string]dm.DaemonManager
 func (engine *DockerTaskEngine) pullContainerManifest(
 	task *apitask.Task, container *apicontainer.Container,
 ) dockerapi.DockerContainerMetadata {
-	// Currently a no-op
-	logger.Debug("Manifest pull is currently a no-op", logger.Fields{
-		field.TaskID:    task.GetID(),
-		field.Container: container.Name,
-		field.Image:     container.Image,
+	switch container.Type {
```

Comment:
Why not use [Container.IsInternal()](https://pkg.go.dev/github.com/aws/amazon-ecs-agent/agent/api/container#Container.IsInternal) function here instead? Are there some "internal" containers that we do actually want to get the digest?

Reply:
I followed the skip logic from `pullContainer`. 
https://github.com/aws/amazon-ecs-agent/blob/b702281f0f6aa2fba5614050769aa337d1456853/agent/engine/docker_task_engine.go#L1251-L1258

I agree that `container.IsInternal()` is a better way to do it. I will make this change.

Reply:
Updated